subsystems/networking/networkmanager_linux.go

```diff
@@ -708,6 +712,7 @@ func (n *Networking) mainLoop(ctx context.Context) {
 
 	scanChan := make(chan bool, 16)
 	inputChan := make(chan userInput, 10)
+	var forceStayInPModeUntilRecvUserInput bool
```

Comment:
I'm not sure I understand the purpose of this variable. It already stays in provisioning mode until it receives input OR the long timeout occurs (10 minutes.) The "ShouldExit" calculation below already includes "userInputReceived" as a condition for exiting. Was this some sort of workaround because you reset the lastTried timestamp on the failed network too early above?